apis/gateway/v1beta1/apirule_types.go

```diff
@@ -74,6 +75,7 @@ type APIRuleStatus struct {
 
 // APIRule is the Schema for ApiRule APIs.
 // +kubebuilder:storageversion
+// +kubebuilder:deprecatedversion:warning=APIRule version v1beta1 is deprecated in favor of v1beta2. You need to migrate to v1beta2. We plan to remove v1beta1 in xx.2023.
```

Comment:
Think you need to adapt it
```suggestion
// +kubebuilder:deprecatedversion:warning=APIRule version v1beta1 is deprecated in favor of v1beta2. You need to migrate to v1beta2. We plan to remove v1beta1 in xx.2024.
```

Reply:
Lets also discuss again in team what end date we would like to have here.